src/interfaces/IHookMetadata.sol

```diff
@@ -0,0 +1,86 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+interface IHookMetadata {
+    /// @notice Represents the auditor 
+    /// @param name The name of the auditor.
+    /// @param uri The URI with additional information about the auditor.
+    /// @param authors List of authors responsible for the audit.
+    struct Auditor {
+        string name;
+        string uri;
+        string[] authors;
+    }
+
+
+    /// @notice Represents a summary of the audit.
+    /// @param auditor The auditor who performed the audit.
+    /// @param issuedAt The timestamp at which the audit was issued.
+    /// @param ercs List of ERC standards that were covered in the audit.
+    /// @param bytecodeHash Hash of the audited smart contract bytecode.
+    /// @param auditHash Hash of the audit document.
+    /// @param auditUri URI with additional information or the full audit report.
+    struct AuditSummary {
+        Auditor auditor;
+        uint256 issuedAt;
+        uint256[] ercs;
+        bytes32 bytecodeHash;
```

Comment:
i do wonder how practical a bytecodeHash is in practice? It changes even you change the optimiser runs, sometimes if you the deploy parameters, and other very small tweaks - and because its not reversible it might become impossible to see what someone _actually_ audited? 
If you have a bytecodeHash from an auditor... how are you meant to find the version of the code that they audited?

Reply:
Just thinking out loud here, it still might be the best option. But just wondering if youve thought this through

Reply:
So I think we are talking about the next options here:

- commit hash - I don't this reasonable because someone can actually change the code during the deployment and you have to read an actual contract code anyway. But would love to hear your opinion;

- bytecodeHash - I think we can also ask devs to deploy with some specific params for optimiser, etc, to be able to allow onchain audit verification. As for deploy parameters - I think we can focus only on the code itself without constructor params;

> how are you meant to find the version of the code that they audited?

So, I think the regular flow - they have to have a link to the repo on their website, commit hash, etc. But as for the onchain verification I see bytecodeHash as only one possible option